lib/diversity/component.rb

```diff
@@ -56,30 +56,26 @@ def initialize(spec, options)
       @options = DEFAULT_OPTIONS.keep_merge(options)
       @logger = @options[:logger]
 
-      schema = JsonSchemaCache[
-                 MASTER_COMPONENT_SCHEMA,
-                 { validate_spec: @options[:validate_spec] }
-               ]
-
       # Handle both Hash and String specs
       @raw = (spec.is_a?(String) ? parse_config(spec) : spec)
 
-      validation = schema.validate(@raw)
-      raise Diversity::Exception, "Configuration is not valid:\n" + validation.join("\n"), caller if
-        validation.length > 0
+      if (@options[:validate_spec])
+        schema = JsonSchemaCache[
+          MASTER_COMPONENT_SCHEMA,
+          { validate_spec: @options[:validate_spec] }
+        ]
+
+        validation = schema.validate(@raw)
+        raise Diversity::Exception,
+              "Configuration is not valid:\n" + validation.join("\n"), caller if
+          validation.length > 0
+      end
 
-      @checksum = Digest::SHA1.hexdigest(dump)
+      @checksum = "#{name}:#{version}"
```

Comment:
I don't agree with this simplification. What you essentially are saying is that *any* object that has the property checksum with this simple string pattern is the same as any other. That's not really a checksum, that's like claiming that anyone calling themselves 'Bob' is the same person.

The point of the original checksum was to be able to differentiate between two components (maybe from different sources) by looking at the *entire* structure. Maybe we don't need to be able to do that, but if so, the need for a checksum member variable  is non-existant. If you insist on doing this, just remove @checksum altogether and define ==(other) as
```ruby
def ==(other)
  return @name == other.name && @version == other.version
end
```